config/socket/game.js

```diff
@@ -1,428 +1,577 @@
       // Verify that the player hasn't previously picked a card
-      var previouslySubmitted = false;
-      _.each(this.table, function(pickedSet, index) {
-        if (pickedSet.player === thisPlayer) {
-          previouslySubmitted = true;
-        }
-      });
-      if (!previouslySubmitted) {
-        // Find the indices of the cards in the player's hand (given the card ids)
-        var tableCard = [];
-        for (var i = 0; i < thisCardArray.length; i++ ) {
-          var cardIndex = null;
-          for (var j = 0; j < this.players[playerIndex].hand.length; j++) {
-            if (this.players[playerIndex].hand[j].id === thisCardArray[i]) {
-              cardIndex = j;
+        let previouslySubmitted = false;
+        _.each(this.table, (pickedSet, index) => {
+          if (pickedSet.player === thisPlayer) {
+            previouslySubmitted = true;
+          }
+        });
+        if (!previouslySubmitted) {
+        // Find the indices of the cards in the player's
+        // hand (given the card ids)
+          const tableCard = [];
+          for (let i = 0; i < thisCardArray.length; i += 1) {
+            let cardIndex = null;
+            for (let j = 0; j < this.players[playerIndex].hand.length; j += 1) {
+              if (this.players[playerIndex].hand[j].id === thisCardArray[i]) {
+                cardIndex = j;
+              }
             }
+            console.log('card', i, 'is at index', cardIndex);
+            if (cardIndex !== null) {
+              tableCard.push(this.players[playerIndex].hand.splice(cardIndex, 1)[0]);
+            }
+            console.log('table object at', cardIndex, ':', tableCard);
           }
-          console.log('card',i,'is at index',cardIndex);
-          if (cardIndex !== null) {
-            tableCard.push(this.players[playerIndex].hand.splice(cardIndex,1)[0]);
+          if (tableCard.length === this.curQuestion.numAnswers) {
+            this.table.push({
+              card: tableCard,
+              player: this.players[playerIndex].socket.id
+            });
+          }
+          console.log('final table object', this.table);
+          if (this.table.length === this.players.length - 1) {
+            clearTimeout(this.choosingTimeout);
+            this.stateJudging(this);
+          } else {
+            this.sendUpdate();
           }
-          console.log('table object at',cardIndex,':',tableCard);
-        }
-        if (tableCard.length === this.curQuestion.numAnswers) {
-          this.table.push({
-            card: tableCard,
-            player: this.players[playerIndex].socket.id
-          });
-        }
-        console.log('final table object',this.table);
-        if (this.table.length === this.players.length-1) {
-          clearTimeout(this.choosingTimeout);
-          this.stateJudging(this);
-        } else {
-          this.sendUpdate();
         }
       }
+    } else {
+      console.log('NOTE:', thisPlayer, 'picked a card during', this.state);
```

Comment:
Why are we logging to the console?

Reply:
This line of code was in the original codebase, It was recorded as part of my changes because I added space after the commas